### PR TITLE
Fixed logic for when to cull entities from having their viewport manipulators regenerated.

### PR DESCRIPTION
**Pull request by cgalvan**
There are some cases where we don't want to show the transform manipulators in the viewport for certain selected entities (focus root, and read-only entities). Previously, this culling was being done in the `RefreshSelectedEntityIds` method, but this only catches selection changes that occur outside of the viewport (e.g. selecting entities in the Entity Outliner). However, this doesn't get called if the selection was changed by the viewport itself (e.g. picking an entity by selecting it in the viewport). So, moved this logic to the `RegenerateManipulators` method so that it gets called everytime, regardless of who modified the selection.

ReadOnlyViewportManipulatorsHidden

Signed-off-by: Chris Galvan <chgalvan@amazon.com>

### CODE REVIEW

**Review by AMZN-daimini — APPROVED** [review]
Thanks :D

**Comment by hultonha** [comment]
Is there any chance we can look at getting a test in-place for this (there are lots of examples in `EditorTransformComponentSelectionTests.cpp`). We can then catch these sorts of issues from happening again.

**Review by NicholasVanSickle — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp [hunk 1]
```diff
@@ -2504,6 +2504,29 @@ namespace AzToolsFramework
...
+        {
+            AzFramework::EntityContextId editorEntityContextId = GetEntityContextId();
+            if (AZ::EntityId focusRoot = prefabFocusPublicInterface->GetFocusedPrefabContainerEntityId(editorEntityContextId);
+                focusRoot.IsValid())
+            {
+                m_selectedEntityIds.erase(focusRoot);
+            }
+        }
+
+        // Do not create manipulators for any entities marked as read only
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp:2518]: uber nit: Can you please use lower case for comments in the .cpp file (matches style of other comments in the file)

Comment (cgalvan) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp]: Sure thing!

**Comment by cgalvan** [comment]
> Is there any chance we can look at getting a test in-place for this (there are lots of examples in `EditorTransformComponentSelectionTests.cpp`). We can then catch these sorts of issues from happening again.

I won't have time to create tests now, but I'll create a backlog item for them.

**Comment by hultonha** [comment]
> > Is there any chance we can look at getting a test in-place for this (there are lots of examples in `EditorTransformComponentSelectionTests.cpp`). We can then catch these sorts of issues from happening again.
> 
> I won't have time to create tests now, but I'll create a backlog item for them.

Is this going into RTE2111? What's the massive rush to have to get this in otherwise?

**Review by hultonha — APPROVED** [review]

**Comment by AMZN-daimini** [comment]
> > > Is there any chance we can look at getting a test in-place for this (there are lots of examples in `EditorTransformComponentSelectionTests.cpp`). We can then catch these sorts of issues from happening again.
> > 
> > 
> > I won't have time to create tests now, but I'll create a backlog item for them.
> 
> Is this going into RTE2111? What's the massive rush to have to get this in otherwise?

Hey Tom, no this is going to development but we were asked to deliver the read-only entity feature by end of year.

**Pull request merged**